### ISSUE TITLE
D3DRS_SRGBWRITEENABLE render state probably not properly managed between different shaders [i.e. in CS:GO]

### ISSUE THREAD

**Issue by ripieces** [ValveSoftware/source-sdk-2013#329]
Hello.

The problem gets most visible, when you overload the materials with a depthdebug shader based material:

This is how it should look (please note that this is work in progress and thus i.e. I didn't account for the 3D skybox scale yet):
![D3DRS_SRGBWRITEENABLE render state fixed](http://i57.tinypic.com/dmtter.jpg)

This how it actually looks:
![D3DRS_SRGBWRITEENABLE render state not fixed](http://i61.tinypic.com/2mrgrxi.jpg)

There are several hints that some elements (i.e. world texture decal shaders and maybe the GUI shaders) enable D3DRS_SRGBWRITEENABLE and others disable it and that in-between the state is "random" and not properly managed.
In other words things like pShaderShadow->EnableSRGBWrite probably affect other shaders, because the state is not properly managed.

Maybe the problem is me because I am using the depth shaders, maybe not.

I can fix it for my self, just wanted to let you know in case you run into a similar problem.

I can't tell from the code if each shader is supposed to default everything or not, if yes, then pShaderShadow->EnableSRGBWrite(false) needs to be added to the debugdepth shader. But I already found hints, that this again might affect other shaders (i.e. in the GUI) that don't set a default either.
